crux-mir/src/Mir/PP.hs

```diff
@@ -235,86 +239,86 @@ instance Pretty Constant where
     pretty (Constant _a b) = pretty b
 
 instance Pretty NullOp where
-    pretty SizeOf = text "sizeof"
-    pretty Box    = text "box"
+    pretty SizeOf = pretty "sizeof"
+    pretty Box    = pretty "box"
 
 instance Pretty BorrowKind where
-    pretty = text . show
+    pretty = pretty . show
 
 
 
 instance Pretty UnOp where
-    pretty Not = text "!"
-    pretty Neg = text "-"
+    pretty Not = pretty "!"
+    pretty Neg = pretty "-"
 
 instance Pretty BinOp where
     pretty op = case op of
-      Add -> text "+"
-      Sub -> text "-"
-      Mul -> text "*"
-      Div -> text "/"
-      Rem -> text "%"
-      BitXor -> text "^"
-      BitAnd -> text "&"
-      BitOr -> text "|"
-      Shl -> text "<<"
-      Shr -> text ">>"
-      Beq -> text "=="
-      Lt -> text "<"
-      Le -> text "<="
-      Ne -> text "!="
-      Ge -> text ">="
-      Gt -> text ">"
-      Offset -> text "Offset"
+      Add -> pretty "+"
+      Sub -> pretty "-"
+      Mul -> pretty "*"
+      Div -> pretty "/"
+      Rem -> pretty "%"
+      BitXor -> pretty "^"
+      BitAnd -> pretty "&"
+      BitOr -> pretty "|"
+      Shl -> pretty "<<"
+      Shr -> pretty ">>"
+      Beq -> pretty "=="
+      Lt -> pretty "<"
+      Le -> pretty "<="
+      Ne -> pretty "!="
+      Ge -> pretty ">="
+      Gt -> pretty ">"
+      Offset -> pretty "Offset"
 
 instance Pretty CastKind where
-    pretty = text . show
+    pretty = pretty . show
 
 instance Pretty IntLit where
-  pretty i = text $ case i of
-    U8 i0 -> show i0
-    U16 i0 -> show i0
-    U32 i0 -> show i0
-    U64 i0 -> show i0
-    U128 i0 -> show i0
-    Usize i0 -> show i0
-    I8 i0 -> show i0
-    I16 i0 -> show i0
-    I32 i0 -> show i0
-    I64 i0 -> show i0
-    I128 i0 -> show i0
-    Isize i0 -> show i0
+  pretty i = pretty $ case i of
+    U8 i0 -> i0
+    U16 i0 -> i0
+    U32 i0 -> i0
+    U64 i0 -> i0
+    U128 i0 -> i0
+    Usize i0 -> i0
+    I8 i0 -> i0
+    I16 i0 -> i0
+    I32 i0 -> i0
+    I64 i0 -> i0
+    I128 i0 -> i0
+    Isize i0 -> i0
 
 instance Pretty FloatLit where
-  pretty = text . show
+  pretty = pretty . show
 
 
 instance Pretty Substs where
   pretty (Substs b) = langle <> hcat (punctuate comma (map pretty b)) <> rangle
-  
+
 instance Pretty ConstVal where
     pretty (ConstFloat i)   = pretty i
     pretty (ConstInt i)     = pretty i
-    pretty (ConstStr i)     = char '\"' <> text (show i) <> char '\"'
-    pretty (ConstByteStr i) = text (show i)
+    pretty (ConstStr i)     = pretty '\"' <> pretty (show i) <> pretty '\"'
```

Comment:
Maybe `pretty (ConstStr i) = dquotes (pretty i)` and `pretty (ConstByteStr i) = pretty i`

Simplifies the quoting, and gets rid of the show on the internal (if possible).

Reply:
I would have combined the `pretty` and `show` on those if I had known I could. The `i` on both of those constructors has type `ByteString`, and I actually don't know exactly how `Show` instance is defined for that, and I didn't check to see if there was already a `Pretty` instance. If there is, and they match, then I could go back and combine those.

Reply:
I just looked through some haddocks. The `prettyprinter` library doesn't provide a `Pretty` instance for `ByteString`. The `Show` instance for `ByteString` is completely undocumented, but from peeking at the source, `show` is the same as `show . Data.ByteString.Char8.unpack`, so it's like printing a `String` where every `Char` fits in 8 bits.

So this code seems a bit weird, then. As far as I can tell, a `ConstByteStr` will print out as a quoted string (just like calling `show` on a `String`) and a `ConstStr` will print out as a string with double-double quotes around it, like `""foo""`. I don't know if this is what was intended here or not.